packages/admin/cms-admin/src/form/file/FileUploadField.tsx

```diff
@@ -0,0 +1,16 @@
+import { Field, FieldProps } from "@comet/admin";
+import React from "react";
+
+import { FinalFormFileUpload, FinalFormFileUploadProps } from "./FinalFormFileUpload";
+import { GQLFinalFormFileUploadFragment } from "./FinalFormFileUpload.generated";
+
+export type FileUploadFieldProps<MaxFiles> = (MaxFiles extends 1
+    ? { maxFiles?: MaxFiles } & FieldProps<GQLFinalFormFileUploadFragment, HTMLInputElement>
+    : { maxFiles?: MaxFiles } & FieldProps<GQLFinalFormFileUploadFragment[], HTMLInputElement>) &
```

Comment:
Shouldn't this be based on multiple instead?

Reply:
Yes, changed in 35a5d1d78abc01450deb97b3f9f71e3478174bf7